lib/oslc_lisp.pl

```
@@ -24,10 +24,23 @@
   oslc_client:post_resource(ResourceIRI, PostURI, []).
 
 lisp:func(send, [ResourceIRI, PostURI, Options], true) :- !,
-  oslc_client:post_resource(ResourceIRI, PostURI, Options).
+  map_content_type(Options, Options1),
+  oslc_client:post_resource(ResourceIRI, PostURI, Options1).
 
 lisp:func(send_graph, [GraphIRI, PostURI], true) :- !,
   oslc_client:post_graph(GraphIRI, PostURI, []).
 
 lisp:func(send_graph, [GraphIRI, PostURI, Options], true) :- !,
-  oslc_client:post_graph(GraphIRI, PostURI, Options).
+  map_content_type(Options, Options1),
+  oslc_client:post_graph(GraphIRI, PostURI, Options1).
+
+map_content_type(Options, Options1) :-
+  option(content_type(ContentType), Options),
```

@morlic I think, you need to put option/2 in if instead of var as it will fail silently if option doesn't exist.

---

option/3 you mean?

---

No, option/2 fails if option is not found.